core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -148,9 +150,15 @@ public void close() throws IOException {
         // The dst can exist because of a master failure during commit. This can only happen
         // when the primary master starts. We can delete either the src or dst. We delete dst and
         // do rename again.
-        mUfs.deleteFile(dst);
+        if (!mUfs.deleteFile(dst)) {
+          LOG.warn("Failed to delete duplicate journal log file {}", dst);
+          return;
```

Comment:
Why do we return here instead of continuing with the rename (and risk failing) but we continue in `recoverLastPersistedJournalEntry`?

Reply:
removed the return, might as well give rename a shot

Reply:
I think we could definitely make that work, but I'd prefer to write a simple boilerplate class over having to deal with a `Spy`. Reading and debugging is much more straightforward, especially with the ability to step through and see what's happening.